aiorun.py

```diff
@@ -62,12 +67,13 @@ def shutdown_waits_for(coro, loop=None):
         """
         try:
             result = await coro
-            try:
-                fut.set_result(result)
-            except asyncio.InvalidStateError:
-                logger.warning('Failed to set result.')
         except (CancelledError, Exception) as e:
-            fut.set_exception(e)
+            dispatch = partial(fut.set_exception, e)
```

Comment:
Instead of `dispatch` change the identifier to something like `update_future_result` or something like that.

Reply:
update_future_result isn't true for the set_exception branch

Reply:
going with set_fut_done, to match the Future.done method

Reply:
good choice 👍